src/ws.cpp

```diff
@@ -546,6 +599,50 @@ void Instance::foreachDmaBufModifier(std::function<void (int format, uint64_t mo
     }
 }
 
+void Instance::initializeVideoPlaneDisplayDmaBuf(VideoPlaneDisplayDmaBufCallback updateCallback, VideoPlaneDisplayDmaBufEndOfStreamCallback endOfStreamCallback)
+{
+    if (m_videoPlaneDisplayDmaBuf.object)
+        return;
+
+    m_videoPlaneDisplayDmaBuf.object = wl_global_create(m_display, &wpe_video_plane_display_dmabuf_interface, 1, this,
+        [](struct wl_client* client, void*, uint32_t version, uint32_t id)
+        {
+            struct wl_resource* resource = wl_resource_create(client, &wpe_video_plane_display_dmabuf_interface, version, id);
+            if (!resource) {
+                wl_client_post_no_memory(client);
+                return;
+            }
+
+            wl_resource_set_implementation(resource, &s_wpeDmaBufInterface, nullptr, nullptr);
+        });
+    m_videoPlaneDisplayDmaBuf.updateCallback = updateCallback;
+    m_videoPlaneDisplayDmaBuf.endOfStreamCallback = endOfStreamCallback;
+}
+
+void Instance::handleVideoPlaneDisplayDmaBuf(struct wpe_video_plane_display_dmabuf_export* dmabuf_export, uint32_t id, int fd, int32_t x, int32_t y, int32_t width, int32_t height, uint32_t stride)
+{
+    if (!m_videoPlaneDisplayDmaBuf.updateCallback) {
+        if (fd >= 0)
+            close(fd);
+        return;
+    }
+
+    (m_videoPlaneDisplayDmaBuf.updateCallback)(dmabuf_export, id, fd, x, y, width, height, stride);
```

Comment:
The parentheses around `m_videoPlaneDisplayDmaBuf.updateCallback` are not needed, the following is fine:

```cpp
m_videoPlaneDisplayDmaBuf.updateCallback(dmabuf_export, id, fd, x, y, width, height, stride);
```

Either do that, or explicitly dereference the pointer when using parentheses:

```cpp
(*m_videoPlaneDisplayDmaBuf.updateCallback)(dmabuf_export, id, fd, x, y, width, height, stride);
```

(C++ and modern C don't require to explicitly dereference function pointers to invoke the function being pointed to, so I would rather go with removing the parentheses.)